.github/workflows/pull-request.yml

```diff
@@ -86,50 +104,58 @@ jobs:
             "VERSION=${{ env.TAG }}"
             "COMMIT=${{ github.SHA }}"
             "RKE_VERSION=${{ steps.env.outputs.RKE_VERSION }}"
-            "ARCH=${{ matrix.arch }}"
+            "ARCH=${{ env.ARCH }}"
             "CATTLE_RANCHER_WEBHOOK_VERSION=${{ steps.env.outputs.CATTLE_RANCHER_WEBHOOK_VERSION }}"
             "CATTLE_CSP_ADAPTER_MIN_VERSION=${{ steps.env.outputs.CATTLE_CSP_ADAPTER_MIN_VERSION }}"
             "CATTLE_FLEET_VERSION=${{ steps.env.outputs.CATTLE_FLEET_VERSION }}"
-          tags: ${{ env.IMAGE }}:${{ env.TAG }}-${{ matrix.arch }}
+          tags: ${{ env.IMAGE }}:${{ env.TAG }}-${{ env.ARCH }}
           context: .
-          platforms: "${{ matrix.os }}/${{ matrix.arch }}"
+          platforms: "${{ matrix.os }}/${{ env.ARCH }}"
           file: ./package/Dockerfile
           labels: "${{ steps.meta.outputs.labels }}"
-          outputs: type=docker,dest=/tmp/rancher-${{ matrix.os }}-${{ matrix.arch }}.tar
+          outputs: type=docker,dest=/tmp/rancher-${{ matrix.os }}-${{ env.ARCH }}.tar
       - name: Upload image
         uses: actions/upload-artifact@v4
         with:
-          name: "rancher-${{ matrix.os }}-${{ matrix.arch }}"
-          path: /tmp/rancher-${{ matrix.os }}-${{ matrix.arch }}.tar
+          name: "rancher-${{ matrix.os }}-${{ env.ARCH }}"
+          path: /tmp/rancher-${{ matrix.os }}-${{ env.ARCH }}.tar
           if-no-files-found: error
           retention-days: 4
           overwrite: false
   build-agent:
     needs: [build-server]
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         os: [linux]
-        arch: [amd64, arm64]
+        arch: [x64, arm64]
+    runs-on: runs-on,runner=4cpu-${{ matrix.os }}-${{ matrix.arch }},run-id=${{ github.run_id }}
+    container: registry.suse.com/bci/bci-base:latest
```

Comment:
Are we not meant to use non-containerised runners for this?

Reply:
Updated to drop `container`